core/src/main/scala/matryoshka/Nu.scala

```diff
@@ -20,25 +20,28 @@ import Recursive.ops._
 
 import scalaz._, Scalaz._
 
-final case class Nu[F[_]](unNu: Id ~> F, a: Id[_])
+/** This is for coinductive (potentially infinite) recursive structures, models
+  * the concept of “codata”, aka, the “greatest fixed point”.
+  */
+sealed abstract class Nu[F[_]]
+private final case class Nu0[F[_], A](unNu: A => F[A], a: A) extends Nu[F]
```

Comment:
@sellout I'd argue the version with the type member is a more direct way of encoding this in scala (both are almost exactly the same code if you squint, with `Nu.apply == Nu0`).

You could rewrite `Nu.apply` to be `def apply[F[_], B](b: B)(f: B => F[B]): Nu[F]` to make it really look like `ana`.